pkgs/development/python-modules/gpt-2-simple/default.nix

```
@@ -15,8 +15,6 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ regex requests tqdm numpy toposort tensorflow ];
 
-  dontCheck = true; # no tests in upstream
```

this breaks the build (tested x86_64-linux)

please:
```suggestion

  doCheck = false; # no tests in upstream
```

---

✔️ Applied!

Though I'd like to understand why the change broke the build for `gpt-2-simple`, @ckiee. What was the error message? Mind posting a build log?

`doCheck` should be false in [`pkgs/development/interpreters/python/mk-python-derivation.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix#L147) as [`doCheckByDefault`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/config.nix#L53) uses [`mkMassRebuild`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/config.nix#L22), which defaults to `false`.

As the GitHub checks also succeeded without error (maybe that error isn't caught?), could the build breakage be related to something in your Nix environment?

What else might I be missing here?

---

> `doCheck` should be false in [`pkgs/development/interpreters/python/mk-python-derivation.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix#L147) as [`doCheckByDefault`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/config.nix#L53) uses [`mkMassRebuild`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/config.nix#L22), which defaults to `false`.

`doCheck` is an **argument** of `buildPythonPackage` and `buildPythonApplication`. These two build helpers are mainly implemented by `mk-python-derivation.nix`, and is decorated (to add `.overridePythonAttrs`, etc.) in `python-packages-base.nix`.

These build helpers sets the `doCheck` **derivation attribute** of the result package `false` via the `doCheck` argument of `stdenv.mkDerivation` build helper. In the mean time the `doInstallCheck` derivation attribute is set according to the `doCheck` argument taken by `buildPython*` build helpers (via the `doInstallCheck` argument of `stdenv.mkDerivation`).

> As the GitHub checks also succeeded without error (maybe that error isn't caught?), could the build breakage be related to something in your Nix environment?

The build with install checks enabled fails to build on my `x86_64-linux` NixOS laptop. The build is done inside the Nix sandbox. Environment configuration has no effect on a pure and sandboxed build. Did OfBorg test the build of `python3Packages.gpt-2-simple` without `doCheck = false`? At least current test result doesn't include `python3Packages.gpt-2-simple`. (I haven't dug into how OfBorg decide if a package should be build as part of the tests.)

---

> `doCheck` should be false in [`pkgs/development/interpreters/python/mk-python-derivation.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix#L147) as [`doCheckByDefault`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/config.nix#L53) uses [`mkMassRebuild`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/config.nix#L22), which defaults to `false`.

The way `mk-python-derivation.nix` handles this build helper argument is confusing. It seems to default it `false`, but effectively [default it to `true`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix#L312) when specifying the `doInstallCheck` derivation attribute.

---

Thanks for the added context, @ShamrockLee, that's very helpful!